lua/autorun/server/sv_cfc_logging_lib.moon

```diff
@@ -1,25 +1,57 @@
-export CFCAlertLib
+class RankManager
+    new: =>
+        -- Generate ordered ulx groups
+        flatten = (list, flattened) ->
+            flattened or= {}
+
+            for k, v in pairs list
+                table.insert flattened, k
+                return flatten v, flattened
+
+            flattened
+
+        ulx_groups = ULib.ucl.getInheritanceTree!
+
+        @ordered_groups = [rank for _,rank in ulx_groups]
```

Comment:
Does this need to use `pairs`?

Reply:
I think I could also do `[rank for rank in *ulx_groups]`, but yes